shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -140,48 +141,52 @@ impl RedisSinkCluster {
                 };
                 Box::pin(one_rx)
             }
+            // Send to the single channel and return its response.
             1 => {
                 let channel = channels.get(0).unwrap();
                 let one_rx = self.choose_and_send(channel, message).await?;
                 Box::pin(one_rx.map_err(|_| anyhow!("no response from single channel")))
             }
+            // Send to all senders.
+            // If any of the responses were a failure then return that failure.
+            // Otherwise return the first successful result
             _ => {
                 let responses = FuturesUnordered::new();
 
                 for channel in channels {
                     responses.push(self.choose_and_send(&channel, message.clone()).await?);
                 }
 
-                // Reassemble upstream responses in any order into an array, as the downstream response.
-                // TODO: Improve error messages within the reassembled response. E.g. which channel failed.
-
                 Box::pin(async move {
                     let response = responses
-                        .fold(vec![], |mut acc, response| async move {
-                            match response {
-                                Ok(Response {
-                                    response: Ok(mut messages),
-                                    ..
-                                }) => acc.push(messages.pop().map_or(
-                                    RedisFrame::Null,
-                                    |mut message| match message.frame().unwrap() {
-                                        Frame::Redis(frame) => frame.take(),
-                                        _ => unreachable!(),
-                                    },
-                                )),
-                                Ok(Response {
-                                    response: Err(e), ..
-                                }) => acc.push(RedisFrame::Error(e.to_string().into())),
-                                Err(e) => acc.push(RedisFrame::Error(e.to_string().into())),
+                        .fold(None, |acc, response| async move {
```

Comment:
My thinking is that in the follow up work any commands that are expected to return unique values will be marked as unusable and shotover will return an error instead of attempting to handle them. e.g. https://redis.io/commands/time/
but this will be determined on a case-by-case basis, for some commands it might be worth pretending that they return the same value.